sample-ios/Sample/HackerNew Post Detail/HackerPostDetailView.swift

```diff
@@ -8,40 +8,48 @@
 
 import Foundation
 import SwiftUI
+import SampleCore
 
 struct HackerPostDetailView: View {
-    @StateObject var viewState: HackerPostDetailViewState
-
+    
+    @StateObject  var viewModel: ObservableViewModel<HackerPostDetailViewState, HackerPostDetailAction, HackerPostDetailViewModel>
+    
     var body: some View {
-        if let errorViewModel = viewState.errorViewModel {
-            FullScreenErrorView(viewModel: errorViewModel)
-        } else if let item = viewState.item {
-            ScrollView(.vertical, showsIndicators: false) {
-                VStack {
-                    Text(item.date)
-                        .foregroundColor(Color.theme.primary)
-                        .font(.body)
-                        .padding(.vertical, 8.0)
-                    Text(item.title)
-                        .foregroundColor(Color.theme.secondary)
-                        .font(.title)
-                        .padding(.bottom, 8.0)
-                    Text(item.author)
-                        .foregroundColor(Color.theme.primary)
-                        .font(.body)
-                        .padding(.bottom, 8.0)
-                    Text(item.text)
-                    Spacer()
-                }
-                .padding(.horizontal, 16.0)
-            }
-            .clipped()
+        switch viewModel.viewState {
+        case is HackerPostDetailViewState.Loading:
+            LoadingOverlayView()
+        case let content as HackerPostDetailViewState.Content:
+            hackerPostDetail(post: content.post)
+        case let error as HackerPostDetailViewState.Error:
+            FullScreenErrorView(viewModel: ErrorViewModel(message: error.message, action: {
+                viewModel.onAction(action: HackerPostDetailAction.Refresh())
+            }))
+        default:
+            unknownViewStateError()
```

Comment:
As already discussed in the Kotlin Multiplatform meeting. This, unfortunately, is not ideal and defeats in a way the purpose of MVI. We need our states explicit and compile safe.

For now, we can go ahead with this, but we need to tackle this again in the near future.